### PR TITLE
Node to node Public Posts

### PR DESCRIPTION
**Pull request by derekdowling**
Okay, this gives us node-to-node public posts in our own api calls. It adds all the necessary infrastructure to integrate more nodes into our system.

### CODE REVIEW

File: api/integrations/aggregator.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
+from ..models import Node
+from . import Integrator
+
+class Aggregator(object):
+
+    @staticmethod
+    def get_integrators():
+        nodes = Node.objects.filter(integrated=True)
+        print nodes
```
Comment (xkb) [inline api/integrations/aggregator.py]: oops

File: api/integrations/integrator.py [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+    def get_author_view(self, author_id):
+        """
+        Combines together a bunch of foreign author calls to one object to help the
+        frontend. Includes, author's info, friends, and posts.
+        """
+        author = get_author(author_id)
+        posts = get_author_posts(author_id)
+        author["posts"] = posts
+        author["friends"] = friends
```
Comment (xkb) [inline api/integrations/integrator.py:94]: shouldn't this be commented out?

Comment (derekdowling) [inline api/integrations/integrator.py:94]: It's not being tested yet.

File: nodes.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
+#!/usr/bin/env python
```
Comment (derekdowling) [inline nodes.py:1]: This is so we don't lose credentials every time we blow away sqlite.

File: api/views/content.py [hunk 4]
```diff
@@ -144,5 +145,14 @@ class PublicPostsViewSet(
...
 
-    def get_queryset(self):
-        return self.queryset.filter(visibility="PUBLIC")
+    def list(self, request):
+        internal_posts = Post.objects.filter(visibility="PUBLIC")
+        serializer = PostSerializer(internal_posts, many=True)
+        posts = serializer.data
+
+        # dont return public posts of other nodes in node-to-node calls
+        if request.auth is not None and request.user.type is not "Node":
```
Comment (xkb) [inline api/views/content.py]: If the user isn't authenticated then request.user is an insttance of `django.contrib.auth.models.AnonymousUser` which should still return public posts of other nodes.

File: api/tests/test_integrations.py [hunk 5]
```diff
@@ -0,0 +1,19 @@
+from ..integrations import Integrator, Accumulator
+from django.test import TestCase
+
+class IntegrationTests(TestCase):
+
+    def setUp(self):
+        self.integrators = Accumulator.get_integrators()
+
+    def test_public_posts(self):
+        print "WTF"
```
Comment (xkb) [inline api/tests/test_integrations.py]: one more

**Comment by xkb** [comment]
LGTM :+1:

**Pull request merged**